src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingInput/StakingInput.tsx

```diff
@@ -21,58 +18,14 @@ const validationSchema = object({
 }).defined();
 
 export type StakingWidgetValues = InferType<typeof validationSchema>;
-type StakeMotionPayload = Action<ActionTypes.MOTION_STAKE>['payload'];
-
-export const getStakingTransformFn = (
-  remainingToStake: string,
-  userMinStake: string,
-  userAddress: string,
-  colonyAddress: string,
-  motionId: string,
-  vote: number,
-) =>
-  mapPayload(({ amount: sliderAmount }) => {
-    const finalStake = getStakeFromSlider(
-      sliderAmount,
-      remainingToStake,
-      userMinStake,
-    );
-
-    return {
-      amount: finalStake,
-      userAddress,
-      colonyAddress,
-      motionId: BigNumber.from(motionId),
-      vote,
-    } as StakeMotionPayload;
-  });
+export type StakingInputRef =
+  | { refetchStakingSlider: ApolloRefetch }
+  | undefined;
 
 const StakingInput = () => {
-  // const handleSuccess = (_, { setFieldValue, resetForm }) => {
-  //     resetForm({});
-  //     setFieldValue('amount', 0);
-  //     scrollToRef?.current?.scrollIntoView({ behavior: 'smooth' });
-  //   },
-
-  const { user } = useAppContext();
-  const { colony } = useColonyContext();
-  const {
-    motionId,
-    remainingStakes: [nayRemaining, yayRemaining],
-    userMinStake,
-    canBeStaked,
-    isObjection,
-  } = useStakingWidgetContext();
-  const remainingToStake = isObjection ? nayRemaining : yayRemaining;
-  const vote = isObjection ? MotionVote.Nay : MotionVote.Yay;
-  const transform = getStakingTransformFn(
-    remainingToStake,
-    userMinStake,
-    user?.walletAddress ?? '',
-    colony?.colonyAddress ?? '',
-    motionId,
-    vote,
-  );
+  const stakingInputRef = useRef<StakingInputRef>();
```

Comment:
In my mind, this is the best way to pass data from `StakingSlider` up the component hierarchy (using `useImperativeHandler in `useStakingSlider`). It's reused in RaiseObjectionDialog, meaning we can't use lifted state or context.